Mathlib/Algebra/Lie/Semisimple.lean

```diff
@@ -120,31 +96,453 @@ theorem hasTrivialRadical_iff_no_abelian_ideals :
     HasTrivialRadical R L ↔ ∀ I : LieIdeal R L, IsLieAbelian I → I = ⊥ := by
   rw [hasTrivialRadical_iff_no_solvable_ideals]
   constructor <;> intro h₁ I h₂
-  · haveI : IsLieAbelian I := h₂; apply h₁; exact LieAlgebra.ofAbelianIsSolvable R I
-  · haveI : IsSolvable R I := h₂; rw [← abelian_of_solvable_ideal_eq_bot_iff]; apply h₁
+  · apply h₁; exact LieAlgebra.ofAbelianIsSolvable R I
+  · rw [← abelian_of_solvable_ideal_eq_bot_iff]; apply h₁
     exact abelian_derivedAbelianOfIdeal I
 #align lie_algebra.is_semisimple_iff_no_abelian_ideals LieAlgebra.hasTrivialRadical_iff_no_abelian_ideals
 
 @[simp]
-theorem center_eq_bot_of_hasTrivialRadical [h : HasTrivialRadical R L] : center R L = ⊥ := by
-  rw [hasTrivialRadical_iff_no_abelian_ideals] at h; apply h; infer_instance
-#align lie_algebra.center_eq_bot_of_semisimple LieAlgebra.center_eq_bot_of_hasTrivialRadical
+theorem HasTrivialRadical.center_eq_bot [HasTrivialRadical R L] : center R L = ⊥ :=
+  HasTrivialRadical.eq_bot_of_isSolvable _
+#align lie_algebra.center_eq_bot_of_semisimple LieAlgebra.HasTrivialRadical.center_eq_bot
+
+/-- A Lie algebra is simple if it is irreducible as a Lie module over itself via the adjoint
+action, and it is non-Abelian. -/
+class IsSimple : Prop where
+  eq_bot_or_eq_top : ∀ I : LieIdeal R L, I = ⊥ ∨ I = ⊤
+  non_abelian : ¬IsLieAbelian L
+#align lie_algebra.is_simple LieAlgebra.IsSimple
+
```

Comment:
I think we should probably have some more namespacing
```suggestion

namespace IsSimple

variable [IsSimple R L]

```
and close just above `SemisimpleGenerators`